operations/helm/charts/mimir-distributed/templates/ingester/ingester-statefulset.yaml

```diff
@@ -52,8 +52,17 @@ spec:
     metadata:
       labels:
         {{- include "mimir.podLabels" $args | nindent 8 }}
+        grafana.com/no-downscale: {{$rolloutZone.noDownscale}}
```

Comment:
These two annotations are mutually exclusive:
* A statefulset can either be "no-downscale" in which case, it's not allowed to be scaled down by anything.
* Or it can be scaled down by the rollout-operator

Reply:
sorry, I was interfering. I think having the prepare-scaledown and graceful scaledown procedure by the rollout operator are easier to reason about from an operator's POV - you just scale the StatfulSets as you would normally and the rollout-operator sorts out preparations and wait times.

But I suspect Nick's point was that setting both doesn't make sense. So it would be useful if we add some validations that they both cannot be set. Perhaps you can add a named template (helm function) to [this file](https://github.com/grafana/mimir/blob/8f2763786bc333dbc10b5c38343bacf2a3722871/operations/helm/charts/mimir-distributed/templates/validate.yaml#L263-L262) which checks the values of mimir.podLabels and mimir.podAnnotations and fails the chart rendering when both are set.

Reply:
oh i see, i didn't know about this possibility!

Reply:
sorry I didn't make this clear - I think it's better if we have the prepare-downscale and the downscale leader on by default. This would remove some toil from operators as opposed to only preventing accidental downscaling.